test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetIVsApiTests/NuGetPackageSigningTestCase.cs

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Test.Apex.VisualStudio.Solution;
+using NuGet.StaFact;
+using NuGet.Test.Utility;
+using Test.Utility.Signing;
+using Xunit;
+
+namespace NuGet.Tests.Apex
+{
+    public class NuGetPackageSigningTestCase : SharedVisualStudioHostTestClass, IClassFixture<SignedPackagesTestsApexFixture>
+    {
+        private SignedPackagesTestsApexFixture _fixture;
+
+        public NuGetPackageSigningTestCase(SignedPackagesTestsApexFixture apexSigningFixture)
+            : base(apexSigningFixture)
+        {
+            _fixture = apexSigningFixture;
+        }
+
+        [CIOnlyNuGetWpfTheory]
+        [InlineData(ProjectTemplate.ClassLibrary)]
+        [InlineData(ProjectTemplate.NetCoreConsoleApp)]
+        [InlineData(ProjectTemplate.NetStandardClassLib)]
+        public void InstallSignedPackageFromPMC(ProjectTemplate projectTemplate)
+        {
+            var signedPackage = _fixture.SignedTestPackage;
+
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Arrange
+                EnsureVisualStudioHost();
+                var solutionService = VisualStudio.Get<SolutionService>();
+
+                solutionService.CreateEmptySolution("TestSolution", pathContext.SolutionRoot);
+                var project = solutionService.AddProject(ProjectLanguage.CSharp, projectTemplate, ProjectTargetFramework.V46, "TestProject");
+                project.Build();
+
+                SimpleTestPackageUtility.CreatePackages(pathContext.PackageSource, signedPackage);
+
+                var nugetTestService = GetNuGetTestService();
+                Assert.True(nugetTestService.EnsurePackageManagerConsoleIsOpen());
+
+                var nugetConsole = nugetTestService.GetPackageManagerConsole(project.Name);
```

Comment:
In the other PMC test class I've added helper methods for things such as getting and ensuring the console is open. And for verifying there are no errors in the error window. All of this should be done in one place to make these tests easier to manage. So anything that can be shared between these please do it.

With errors for example, if this fails all you will see is Assert.True was false. It doesn't show you what the errors were, and since this might not repro locally it will be hard to figure out what happened.

I also don't think it makes sense in most scenarios to check the error lists, this is probably going to fail due to build related issues since build is called right before. The extra builds here should also be removed if they are being done after all the asserts, they really have no benefit for the tests.

Reply:
@emgarten the PR for the PMC helper methods you mention is still in WIP. I prefer not to block this PR on that... Either change this class as part of that PR or I can make the change once both are merged.